decryptor/mysql/data_encoder.go

```diff
@@ -0,0 +1,344 @@
+		}
+
+	case TypeTiny:
+		encoded = make([]byte, 1)
+		intValue, err := strconv.ParseInt(string(data), 10, 8)
+		if err != nil {
+			break
+		}
+		err = binary.Write(bytes.NewBuffer(encoded[:0]), binary.LittleEndian, int8(intValue))
+		isNumericType = true
+
+	case TypeShort, TypeYear:
+		encoded = make([]byte, 2)
+		intValue, err := strconv.ParseInt(string(data), 10, 16)
+		if err != nil {
+			break
+		}
+		err = binary.Write(bytes.NewBuffer(encoded[:0]), binary.LittleEndian, int16(intValue))
+		isNumericType = true
+
+	case TypeInt24, TypeLong:
+		encoded = make([]byte, 4)
+		intValue, err := strconv.ParseInt(string(data), 10, 32)
+		if err != nil {
+			break
+		}
+		err = binary.Write(bytes.NewBuffer(encoded[:0]), binary.LittleEndian, int32(intValue))
+		isNumericType = true
+
+	case TypeLongLong:
+		encoded = make([]byte, 8)
+		intValue, err := strconv.ParseInt(string(data), 10, 64)
+		if err != nil {
+			break
+		}
+		err = binary.Write(bytes.NewBuffer(encoded[:0]), binary.LittleEndian, int64(intValue))
+		isNumericType = true
+
+	case TypeFloat:
+		encoded = make([]byte, 4)
+		floatValue, err := strconv.ParseFloat(string(data), 32)
+		if err != nil {
+			break
+		}
+		err = binary.Write(bytes.NewBuffer(encoded[:0]), binary.LittleEndian, float32(floatValue))
+		isNumericType = true
+
+	case TypeDouble:
+		encoded = make([]byte, 8)
+		floatValue, err := strconv.ParseFloat(string(data), 64)
+		if err != nil {
+			break
+		}
+		err = binary.Write(bytes.NewBuffer(encoded[:0]), binary.LittleEndian, float64(floatValue))
+		isNumericType = true
+
+	}
+	if err != nil {
+		//TODO: add error
+		return ctx, data, nil
+	}
+
+	if isNumericType {
+		return ctx, encoded, nil
+	}
+
+	return ctx, PutLengthEncodedString(data), nil
+}
+
+func (p *BaseMySQLDataEncoderProcessor) encodeWithDataType(ctx context.Context, data []byte, setting config.ColumnEncryptionSetting) ([]byte, bool, error) {
+	switch setting.GetEncryptedDataType() {
+	case common.EncryptedType_Bytes:
+		if !base.IsDecryptedFromContext(ctx) {
+			if newValue := setting.GetDefaultDataValue(); newValue != nil {
+				binValue, err := base64.StdEncoding.DecodeString(*newValue)
+				if err != nil {
+					return data, false, err
+				}
+				return PutLengthEncodedString(binValue), true, nil
+			}
+		}
+		return data, false, nil
+	case common.EncryptedType_String:
+		if !base.IsDecryptedFromContext(ctx) {
+			if value := setting.GetDefaultDataValue(); value != nil {
+				return PutLengthEncodedString([]byte(*value)), true, nil
+			}
+		}
+		return PutLengthEncodedString(data), true, nil
+	case common.EncryptedType_Int32:
+		encoded := make([]byte, 4)
+		intValue, err := strconv.ParseInt(string(data), 10, 32)
+		if err != nil {
+			if newVal := setting.GetDefaultDataValue(); newVal != nil {
+				intValue, err = strconv.ParseInt(*newVal, 10, 64)
+				if err != nil {
+					return data, false, err
+				}
+			}
+			// TODO: ask Lagovas what to do in case of error casting to defined type.
```

Comment:
do nothing. return value inappropriate to type as it would be with encrypted binary data in case of expected integer.
at the future we will return DB specific error using wire protocol error notifications.

Reply:
Discussed to implement rollaback type logic